### DISCUSSION TITLE
Configuration file and function

### DISCUSSION THREAD

**Discussion by f-bader** [maester365/maester#743] [Ideas]
Create a JSON that allows for environmental configurations
Create functions that will query the JSON and return the objects

**Comment by fflaten** [comment]
Not `.psd1`? 🙂

**Comment by lbrauns** [comment]
What configurations would be needed in there? I would work on this issue, my use case is to define the emergency access account(s) for the tenant. 

What format should be preferred? JSON would be a possibility for sure. Personally i have a preference for YAML. PSD1 would also be a suitable format. 

@merill maybe?

**Comment by merill** [comment]
@FriedrichWeinmann's PSFramework has a really good configuration system. https://psframework.org/documentation/documents/psframework/configuration.html

It already supports exporting/importing to .psd1, clixml and json. https://psframework.org/documentation/documents/psframework/configuration/persistence-serialization.html

I'm thinking we can leverage that instead of re-inventing the wheel.
